src/common/components/ChangeLog.tsx

```diff
@@ -0,0 +1,160 @@
+  color: ${props => props.theme.changelogs.accentTextColor};
+`;
+
+const defaultPropertyAccessor = (
+  change: unknown,
+  propertyName: string,
+): string | number | Record<string, unknown> | boolean => {
+  if (isObject(change)) {
+    const value = change[propertyName];
+
+    if (isStringOrNumber(value)) {
+      return value;
+    }
+
+    if (isObject(value)) {
+      return value;
+    }
+
+    if (value === null) {
+      return false;
+    }
+
+    throw Error(`The resulting value from change[${propertyName}] is not a string, number, object, or null`);
+  }
+
+  throw Error('Given change parameter is not an object');
+};
+
+const defaultGetChangeItemAction = (action: string): string => {
+  switch (action) {
+    case '+':
+      return 'create';
+    case '~':
+      return 'update';
+    case '-':
+      return 'delete';
+    default:
+      throw Error("Given action does not equal '+', '~', or '-'");
+  }
+};
+
+export const ChangeLog: FC<Props> = ({
+  changeList,
+  totalChanges,
+  changePropertyAccessor = defaultPropertyAccessor,
+  getChangeItemAction = defaultGetChangeItemAction,
+  handleShowAllChanges,
+}) => {
+  const getActionInPastTense = (action: string): string => {
+    return `${action.charAt(0).toUpperCase()}${action.substring(1)}d`;
+  };
+
+  const getShortenedName = (fullName: string): string => {
+    const nameArray: string[] = fullName.split(' ');
+
+    return `${nameArray[0]} ${nameArray[1].charAt(0)}.`;
+  };
+
+  const getDateStr = (date: Date): string => {
+    return date.toLocaleString(undefined, { month: 'short', day: 'numeric' });
+  };
+
+  const getTimeStr = (date: Date): string => {
+    return date.toLocaleString(undefined, { hour: 'numeric', minute: 'numeric' });
+  };
+
+  const getIconForAction = (action: string): IconProp => {
+    switch (action) {
+      case 'create':
+        return 'circle-plus';
+      case 'update':
+        return 'pen';
+      case 'delete':
+        return 'trash-can';
+      default:
+        return 'question'; // if the action isn't recognized
+    }
+  };
+
+  return (
+    <>
+      {changeList.length > 0 ? (
+        <div>
+          <h5 className='mb-3'>Latest Changes</h5>
+          <ListGroup as='ol'>
+            {changeList.map(changeItem => {
+              const id = changePropertyAccessor(changeItem, 'historyId');
+              const changeAction = getChangeItemAction(changePropertyAccessor(changeItem, 'historyType').toString());
+              const creationDate = changePropertyAccessor(changeItem, 'historyDate').toString();
+              const historyUser = changePropertyAccessor(changeItem, 'historyUser');
+
+              let editorName = '';
+
+              if (
+                isObject(historyUser) &&
+                isKeyOfObject('firstName', historyUser) &&
+                isKeyOfObject('lastName', historyUser)
+              ) {
+                editorName = `${historyUser.firstName} ${historyUser.lastName}`;
+              }
```

Comment:
Ditto on this stuff, although this is a good idea for if/when we turn this into a library. It feels a bit [YAGNI](https://www.martinfowler.com/bliki/Yagni.html) at the moment.